olive/engine/engine.py

```diff
@@ -49,6 +52,7 @@ def __init__(
         target: Optional[OliveSystem] = None,
         evaluator_config: Optional[OliveEvaluatorConfig] = None,
         execution_providers: Optional[List[str]] = None,
+        auto_optimizer_config: Optional[AutoOptimizerConfig] = None,
```

Comment:
could we move the auto_optimizer_config out of the engine class? It seems Engine itself doesn't care about the auto_optimizer.

Reply:
As we discussed offline, will refactor this part with accelerator_spec setting up together.